ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx

```diff
@@ -47,11 +47,19 @@ export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}>
     }
 
     private performActionOnSelectedWorkflows(ctx: any, title: string, action: WorkflowOperationAction): Promise<any> {
-        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         let deleteArchived = false;
+        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         if (confirmed) {
             if (title === 'DELETE') {
-                deleteArchived = confirm('Would you also want to delete them in the database?');
+                let hasArchived = false;
+                this.props.selectedWorkflows.forEach((wf: Workflow) => {
+                    if (!hasArchived && isArchivedWorkflow(wf)) {
+                        hasArchived = true;
+                    }
+                });
+                if (hasArchived) {
+                    deleteArchived = confirm('Would you also want to delete them in the database?');
```

Comment:
I'm still not convinced this is the best wording. Consider `Would you also like to delete them from the Workflow Archive database?`

Reply:
Are there any other databases?

Reply:
That seems like implementation detail

Reply:
Depends if your workflow envokes a database or not. If I was a first-time user, or a user that wasn't involved in the setup of Workflows, that error would confuse the heck out of me.